relayer/relayer/src/main.rs

```diff
@@ -139,47 +144,148 @@ async fn main() -> Result<()> {
 
     debug!("Established connection to Ethereum node");
 
-    let eth_listener_config_rc = Arc::clone(&config);
-    let eth_listener_redis_connection_rc = Arc::clone(&redis_connection);
-    let eth_listener_emergency_rc = Arc::clone(&emergency);
+    if let Err(err) = run_listeners(
+        config,
+        azero_connection,
+        azero_signed_connection,
+        eth_connection,
+        eth_signed_connection,
+        redis_connection,
+    )
+    .await
+    {
+        // For most errors we don't want to automatically restart the relayer but wait for manual intervention
+        loop {
+            error!("Error when running listeners, this might require manual investigation or RESTART...");
+            err.chain().enumerate().for_each(|(level, cause)| {
+                let cause = cause.to_string();
+                if cause.len() > 100 {
+                    error!(" {}: {}...", level, &cause[..100]);
+                } else {
+                    error!(" {}: {}", level, cause);
+                }
```

Comment:
Suggestion: print the whole thing. Why are you worried about `cause` being longer than 100?

Reply:
Hm, unfortunate, but maybe we can only shorten that particular one? Better yet - if we wrap it in some error of our own, implement the shortening in there.

Reply:
Changed it to print the whole error once